### DISCUSSION TITLE
Autocomplete from SQL Database

### DISCUSSION THREAD

**Discussion by heejit** [cogentcore/core#1242] [Q&A] [answered]
Is there any widget or a way to autocomplete data from sql database.

I have 10,000+ items in sql database. 

I want user to type few letter on press of Enter key display filtered list of item fro typed letters so user can select the appropriate item.

**Comment by kkoreilly** [comment]
You could fetch all the items and then use a `Chooser` with `SetEditable(true)` and those items; see the [choosers page](https://www.cogentcore.org/core/widgets/inputs/choosers).

**Reply by heejit** [reply]
There are many items and I don't want to fetch all of them at once is there any other way.

Currently in my pyqt app I created a customize widget using textedit which accept sql and when user type and press enter key
it will use text of textedit as argument to sql  and run it and get the result from database and show the result as table in modal Dialog.

this type of widget i need in many forms.

**Reply by kkoreilly** [reply]
Have you considered using [text editors](https://www.cogentcore.org/core/widgets/inputs/text-editors)? You could use that to replicate the same SQL input thing you described above.

**Reply by heejit** [reply]
It is not full sql input just the argument something like this

select id,nm from items where nm like '%TextEditor.Text%'

**Reply by heejit** [reply]
Autocomplete like in E-Commerce websites

**Reply by kkoreilly** [reply, marked as answer]
You could trying using an `Editable` `Chooser` and then change the `Chooser.Items` in `OnInput` based on the current `ch.ChildByName("text-field").(*core.TextField).Text()` using your SQL query.

**Reply by heejit** [reply]
Let me try it
